TaskDispatch.cpp

```diff
@@ -22,15 +25,19 @@ TaskDispatch::TaskDispatch( size_t workers )
     {
         char tmp[16];
         sprintf( tmp, "Worker %zu", i );
-#ifdef __APPLE__
+#ifdef _MSC_VER
+        auto worker = std::thread( [this]{ Worker(); } );
+        System::SetThreadName( worker, tmp );
+#else // Using pthread.
         auto worker = std::thread( [this, tmp]{
+#  ifdef __APPLE__
             pthread_setname_np( tmp );
+#  else // Linux or MinGW.
+            pthread_setname_np( pthread_self(), tmp );
```

Comment:
Some extra `#elif` would be needed to support OpenBSD, FreeBSD and NetBSD, as can be seen in Godot's own pthread name setting interface:
https://github.com/godotengine/godot/blob/a86e7c3bb7d9238e664be8a7c63d1cc19bde0d8c/drivers/unix/thread_posix.cpp#L38-L63

For this PR I kept the changes minimal, but we do try to support \*BSD platforms on a best-effort basis in Godot, so if you're open to that I could add relevant changes to support those (at least theoretically, I don't have current VMs for all three flavors to test the changes). But there might be other issues for \*BSD compatibility so this is properly best left as a potential follow-up once we get to it in Godot.